algorithms/kernel/dtrees/gbt/regression/oneapi/gbt_regression_train_dense_default_oneapi_impl.i

```diff
@@ -0,0 +1,1058 @@
+                    }
+                    else
+                    {
+                        record->isFinalized = true;
+                        record->featureValue = bestSplit._featureValue;
+                        record->featureIdx = bestSplit._featureIndex;
+                        connector.createNode(record->level + 1, record->nid * 2 + 1, nLeft, record->iStart, bestSplit._leftGTotal, bestSplit._leftHTotal, nLeft, par);
+                        connector.createNode(record->level + 1, record->nid * 2 + 2, record->n - nLeft, record->iStart + nLeft, bestSplit._rightGTotal, bestSplit._rightHTotal, nRight, par);
+                        connector.setSplitLevel(record->level + 1);
+                        connector.getSplitNodesMerged(record->nid, splits, false);
+                    }
+                }
+                if (record->nid > 0)
+                {
+                    size_t parentId = (record->nid - 1) / 2;
+                    treeNodeStorages[parentId].clear();
+                }
+            }
+        }
+
+        Collection<TableRecord<algorithmFPType> *> leaves;
+        connector.getLeafNodes(0, leaves);
+        size_t nLeaves = leaves.size();
+
+        for (size_t leafId = 0; leafId < nLeaves; leafId++)
+        {
+            TableRecord<algorithmFPType> *node = leaves[leafId];
+
+            const size_t curRows = node->n;
+            const size_t curOffset = node->iStart;
+
+            algorithmFPType res = 0;
+
+            algorithmFPType val = node->hTotal + par.lambda;
+            if(val != 0.0)
+            {
+                val = -node->gTotal / val;
+                const algorithmFPType inc = val * par.shrinkage;
+
+                res = inc;
+
+                DAAL_CHECK_STATUS_VAR(updateResponse(treeOrder, response, node->iStart, node->n, inc));
+            }
+
+            node->response = res;
+            node->isFinalized = 1;
+        }
+
+        services::Collection<SharedPtr<algorithmFPType> > binValuesHost(nFeatures);
+        DAAL_CHECK_MALLOC(binValuesHost.data());
+        services::Collection<algorithmFPType*> binValues(nFeatures);
+        DAAL_CHECK_MALLOC(binValues.data());
+
+        for (size_t i = 0; i < nFeatures; i++)
+        {
+            binValuesHost[i] = indexedFeatures.binBorders(i).template get<algorithmFPType>().toHost(ReadWriteMode::readOnly);
+            binValues[i] = binValuesHost[i].get();
+        }
+
+        size_t maxLevel = 0;
+        connector.getMaxLevel(0, maxLevel);
+        const size_t nNodes = (1 << (maxLevel + 1)) - 1;
+        const size_t nNodesPresent = connector.getNNodes(0);
+
+        gbt::internal::GbtDecisionTree *pTbl = new gbt::internal::GbtDecisionTree(nNodes, maxLevel, nNodesPresent);
+
+        HomogenNumericTable<double> *pTblImp     = new HomogenNumericTable<double>(1, nNodes, NumericTable::doAllocate);
+        HomogenNumericTable<int>    *pTblSmplCnt = new HomogenNumericTable<int>(1, nNodes, NumericTable::doAllocate);
```

Comment:
May be SyclHomogen .. is better?

Reply:
This table will be used on prediction, which is not implemented on GPU yet.
I will change it to SyclTables as soon as I will add GPU prediction.